walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,123 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.HashSet;
+import java.lang.Math;
 
+/** given a collection of existing meetings (with time ranges and attendees) and a request for 
+  * a meeting (with duration and mandatory/optional attendees), return the times when the meeting 
+  * could happen that day.
+**/
 public final class FindMeetingQuery {
+    private static final TimeRange START_OF_DAY_TIMERANGE =
+        TimeRange.fromStartDuration(TimeRange.START_OF_DAY, 0);
+    private static final TimeRange END_OF_DAY_TIMERANGE =
+        TimeRange.fromStartDuration(TimeRange.END_OF_DAY + 1, 0);
+
+  /** returns time ranges that will accomodate meeting request
+   *  if no time ranges exist all (mandatory + optional) attendees can attend, time ranges that
+   *  accomodate only mandatory attendees is returned instead
+  **/
```

Comment:
What happens if there is not time slots for the mandatory attendees?

Reply:
If there are no time slots available, an empty collection is returned.

Reply:
Can we document that?